File: server/src/models/Discipline.ts
import Model from './Model';
import Teacher from './Teacher';

class Discipline extends Model {
	readonly NumOfLessons: number;

	readonly Teachers: Teacher[];

	readonly NumOfStudents: number;

	constructor(name: string, numOfLessons: number, teachers: Teacher[], numOfStudents: number) {
		super(name);
		this.NumOfLessons = numOfLessons;
		this.Teachers = teachers;
		this.NumOfStudents = numOfStudents;
	}
}

export default Discipline;

File: worker-genetic/models/discipline.py
from typing import List
from .model import Model
from .teacher import Teacher


class Discipline(Model):

    def __init__(self, name: str, lessons_count: int, students_count: int, teachers: List[Teacher]) -> None:
        super().__init__(name)
        self.lessons_count = lessons_count
        self.students_count = students_count
        self.teachers = teachers
